internal/serviceaccount/config.go

```diff
@@ -0,0 +1,229 @@
+			if errors.IsAlreadyExists(err) {
+				fmt.Println("ServiceAccount 'method-service-account' already exists, continuing...")
+				return serviceAccount, nil
+			}
+			return nil, err
+		}
+	}
+	return serviceAccount, nil
+}
+
+func createClusterRole(clientset *kubernetes.Clientset, apply bool) (*rbacv1.ClusterRole, error) {
+	clusterRole := &rbacv1.ClusterRole{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "rbac.authorization.k8s.io/v1",
+			Kind:       "ClusterRole",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "method-read-only-clusterrole",
+		},
+		Rules: []rbacv1.PolicyRule{
+			{
+				APIGroups: []string{""},
+				Resources: []string{"pods", "services", "endpoints", "nodes"},
+				Verbs:     []string{"get", "list", "watch"},
+			},
+			{
+				APIGroups: []string{"apps"},
+				Resources: []string{"deployments", "daemonsets", "replicasets", "statefulsets"},
+				Verbs:     []string{"get", "list", "watch"},
+			},
+			{
+				APIGroups: []string{"batch"},
+				Resources: []string{"jobs", "cronjobs"},
+				Verbs:     []string{"get", "list", "watch"},
+			},
+		},
+	}
+
+	ctx := context.Background()
+	if apply {
+		_, err := clientset.RbacV1().ClusterRoles().Create(ctx, clusterRole, metav1.CreateOptions{})
+		if err != nil {
+			if errors.IsAlreadyExists(err) {
+				fmt.Println("ClusterRole 'method-read-only-clusterrole' already exists, continuing...")
+				return clusterRole, nil
+			}
+			return nil, err
+		}
+	}
+	return clusterRole, nil
+}
+
+func Config(ctx context.Context, k8config *rest.Config, apply bool, namespace string) error {
+	config := k8config
+
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		return err
+	}
+
+	serviceAccountName := "method-service-account"
+
+	clusterRole, err := createClusterRole(clientset, apply)
+	if err != nil {
+		return err
+	}
+	if apply {
+		fmt.Println("- Cluster Role configured (1/4)")
+	} else {
+		fmt.Println("---")
+		prettyPrintYAML(clusterRole)
+	}
+
+	serviceAccount, err := createServiceAccount(clientset, namespace, serviceAccountName, apply)
+	if err != nil {
+		return err
+	}
+	if apply {
+		fmt.Println("- Service Account configured (2/4)")
+	} else {
+		fmt.Println("---")
+		prettyPrintYAML(serviceAccount)
+	}
+
+	secret, err := createSecret(clientset, namespace, serviceAccountName, apply)
+	if err != nil {
+		return err
+	}
+	if apply {
+		fmt.Println("- Secret configured (3/4)")
+	} else {
+		fmt.Println("---")
+		prettyPrintYAML(secret)
+	}
+
+	clusterRoleBinding, err := createClusterRoleBinding(clientset, namespace, serviceAccountName, apply)
+	if err != nil {
+		return err
+	}
+	if apply {
+		fmt.Println("- Cluster Role Binding configured(4/4)")
+		fmt.Println("You are all set! Run 'methodk8 service-account creds' to see your tokens")
+	} else {
+		fmt.Println("---")
+		prettyPrintYAML(clusterRoleBinding)
+	}
+
+	return nil
+}
+
+// prettyPrintYAML prints a Kubernetes object as pretty YAML to the console
+func prettyPrintYAML(obj interface{}) {
```

Comment:
Ah ok, yes, this is basically what I was seeing. What if we instead had all our objects returned into a list so we can apply or print all at once? That would save all these duplicative `if apply .. else` clauses